planning/behavior_path_planner/autoware_behavior_path_planner_common/test/test_static_drivable_area.cpp

```diff
@@ -0,0 +1,541 @@
+    EXPECT_FALSE(result.has_value());
+  }
+  {  // lanes at 0
+    const DrivableLanes l =
+      make_drivable_lanes(make_lanelet({0.0, 1.0}, {5.0, 1.0}, {5.0, -1.0}, {5.0, -1.0}));
+    lanes.push_back(l);
+    const auto result = getOverlappedLaneletId(lanes);
+    EXPECT_FALSE(result.has_value());
+  }
+  {  // lanes at 1, overlap with 0 but ignored since it is the following lane
+    const DrivableLanes l =
+      make_drivable_lanes(make_lanelet({4.0, 1.0}, {8.0, 1.0}, {4.0, -1.0}, {8.0, -1.0}));
+    lanes.push_back(l);
+    const auto result = getOverlappedLaneletId(lanes);
+    EXPECT_FALSE(result.has_value());
+  }
+  {  // lanes at 2, overlap with 1 but ignored since it is the following lane
+    const DrivableLanes l =
+      make_drivable_lanes(make_lanelet({6.0, 1.0}, {10.0, 1.0}, {6.0, -1.0}, {10.0, -1.0}));
+    lanes.push_back(l);
+    const auto result = getOverlappedLaneletId(lanes);
+    EXPECT_FALSE(result.has_value());
+  }
+  {  // lanes at 3, overlap with 1 so 3 is returned
+    const DrivableLanes l =
+      make_drivable_lanes(make_lanelet({5.0, 0.0}, {5.0, 5.0}, {6.0, 0.0}, {6.0, 5.0}));
+    lanes.push_back(l);
+    const auto result = getOverlappedLaneletId(lanes);
+    ASSERT_TRUE(result.has_value());
+    EXPECT_EQ(*result, 3UL);
+  }
+  {  // lanes at 4, overlap with 2 but since 3 overlaps first it is still returned
+    const DrivableLanes l =
+      make_drivable_lanes(make_lanelet({7.0, 0.0}, {7.0, 5.0}, {8.0, 0.0}, {8.0, 5.0}));
+    lanes.push_back(l);
+    const auto result = getOverlappedLaneletId(lanes);
+    ASSERT_TRUE(result.has_value());
+    EXPECT_EQ(*result, 3UL);
+  }
+  {  // change 1 to no longer overlap with 3 and now 4 is the first overlap
+    const DrivableLanes l = make_drivable_lanes(
+      make_lanelet({100.0, 110.0}, {110.0, 100.0}, {100.0, 90.0}, {100.0, 90.0}));
+    lanes[1] = l;
+    const auto result = getOverlappedLaneletId(lanes);
+    ASSERT_TRUE(result.has_value());
+    EXPECT_EQ(*result, 4UL);
+  }
+}
+
+TEST(StaticDrivableArea, cutOverlappedLanes)
+{
+  using autoware::behavior_path_planner::utils::cutOverlappedLanes;
+  tier4_planning_msgs::msg::PathWithLaneId path;
+  std::vector<DrivableLanes> lanes;
+  {  // empty inputs
+    const auto result = cutOverlappedLanes(path, lanes);
+    EXPECT_TRUE(result.empty());
+    EXPECT_TRUE(path.points.empty());
+  }
+  constexpr auto path_size = 10UL;
+  const auto reset_path = [&]() {
+    path.points.clear();
+    for (auto i = 0UL; i < path_size; ++i) {
+      path.points.emplace_back();
+      path.points.back().point.pose.position.x = static_cast<double>(i) * 1.0;
+      path.points.back().point.pose.position.y = 0.0;
+    }
+  };
+  {  // add some path points
+    reset_path();
+    const auto result = cutOverlappedLanes(path, lanes);
+    EXPECT_TRUE(result.empty());
+    ASSERT_EQ(path.points.size(), path_size);
+    for (auto i = 0UL; i < path_size; ++i) {
+      EXPECT_EQ(path.points[i].point.pose.position.x, i * 1.0);
```

Comment:
I think the type of  `position.x` so maybe this might be better?
```suggestion
      EXPECT_DOUBLE_EQ(path.points[i].point.pose.position.x, i * 1.0);
```

Reply:
Thank you. Done: cf0f2e290a